### PR TITLE
Add Option processors

### PR DESCRIPTION
**Pull request by yogevbd**
This PR adds the ability to intercept and modify options each time they applied from any command.
This feature comes in handy when there are guidelines need to be enforced when multiple developers are working together on different modules (or not).

- Add `Navigation.addOptionProcessor(optionPath: string, processor: (value: T, commandName: string) => T)` - When the specific option path is being processed, the processor function gets called and the processor has the option to manipulate this specific option value by returning a new one. It also receives the specific command name which triggered that option processing.
- Fix wallaby

`optionPath` examples: `topBar.title.color`, `bottomTabs.backgroundColor`.

### CODE REVIEW

File: lib/src/commands/OptionsProcessor.ts [hunk 1]
```diff
@@ -12,34 +12,50 @@ import { ColorService } from '../adapters/ColorService';
...
   private processObject(
     objectToProcess: object,
     parentOptions: object,
-    onProcess: (key: string, parentOptions: object) => void
+    onProcess: (key: string, parentOptions: object) => void,
+    commandName: string,
+    path?: string
   ) {
     forEach(objectToProcess, (value, key) => {
+      path = this.resolveObjectPath(key, path);
```
Comment (guyca) [inline lib/src/commands/OptionsProcessor.ts:57]: We should avoid reassigning parameters and enable the [related eslint rule](https://eslint.org/docs/rules/no-param-reassign).

Comment (guyca) [reply lib/src/commands/OptionsProcessor.ts:57]: Seems like the entire class faults in this matter. Lets keep your current implementation and address this in a different PR.

File: lib/src/commands/LayoutTreeCrawler.test.ts [hunk 2]
```diff
@@ -50,28 +50,9 @@ describe('LayoutTreeCrawler', () => {
...
       type: LayoutType.Component,
       data: { name: 'theComponentName', options: {} },
-      children: []
+      children: [],
     };
-    uut.crawl(node);
-    expect(node.data.options).toEqual({ popGesture: true });
-  });
-
-  it('Components: injects options from original component class static property', () => {
```
[the remark this answers is not in the document]
Comment (yogevbd) [reply lib/src/commands/LayoutTreeCrawler.test.ts:59]: it already exists in line 40

File: lib/src/Navigation.ts [hunk 3]
```diff
@@ -22,12 +22,13 @@ import { ColorService } from './adapters/ColorService';
...
 import { AppRegistryService } from './adapters/AppRegistryService';
 import { Deprecations } from './commands/Deprecations';
+import { ProcessorSubscription } from './interfaces/ProcessorSubscription';
 
 export class NavigationRoot {
   public readonly TouchablePreview = TouchablePreview;
 
   private readonly store: Store;
-  private readonly optionProcessorsRegistry: OptionProcessorsRegistry;
+  private readonly optionProcessorsRegistry: OptionProcessorsStore;
```
Comment (guyca) [inline lib/src/Navigation.ts:31]: rename registry to store

**Pull request merged**